src/async_client.rs

```diff
@@ -630,17 +695,37 @@ impl AsyncClient {
                 {
                     if let EventResult::Ok(e) = ephemeral {
                         let mut client = self.base_client.write().await;
-                        client.receive_ephemeral_event(&room_id, e).await;
+                        if client.receive_ephemeral_event(&room_id, e).await {
+                            updated = true;
+                        }
 
                         client.emit_ephemeral_event(room_id, e).await;
                     }
                 }
             }
+
+            if updated {
+                if let Some(store) = self.base_client.read().await.state_store.as_ref() {
+                    if let Some(path) = self.state_store_path.as_ref() {
+                        store
+                            .store_room_state(&path, matrix_room.read().await.deref())
```

Comment:
I think the path should be part of the store, no? E.g. this won't make sense if someone implements a SQLite store. The json store will create subdirs for rooms, a SQLite one won't.